internal/connectors/plugins/public/currencycloud/client/auth.go

```diff
@@ -20,36 +21,40 @@ func (c *Client) authenticate(ctx context.Context, httpClient *http.Client) (str
 	form.Add("login_id", c.loginID)
 	form.Add("api_key", c.apiKey)
 
-	req, err := http.NewRequest(http.MethodPost,
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost,
 		c.buildEndpoint("v2/authenticate/api"), strings.NewReader(form.Encode()))
 	if err != nil {
-		return "", fmt.Errorf("failed to create request: %w", err)
+		return fmt.Errorf("failed to create request: %w", err)
 	}
 
 	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 	req.Header.Add("Accept", "application/json")
 
-	resp, err := httpClient.Do(req)
-	if err != nil {
-		return "", fmt.Errorf("failed to do get request: %w", err)
-	}
-
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK {
-		return "", unmarshalError(resp.StatusCode, resp.Body).Error()
-	}
-
 	//nolint:tagliatelle // allow for client code
 	type response struct {
 		AuthToken string `json:"auth_token"`
 	}
 
 	var res response
-
-	if err = json.NewDecoder(resp.Body).Decode(&res); err != nil {
-		return "", fmt.Errorf("failed to decode response body: %w", err)
+	var errRes currencyCloudError
+	_, err = c.httpClient.Do(req, &res, &errRes)
+	switch err {
+	case nil:
+		c.authToken = res.AuthToken
+	case httpwrapper.ErrStatusCodeUnexpected:
+		// TODO(polo): retryable errors
+		return errRes.Error()
```

Comment:
This appears to have snuck back in after the rebase. Let's return the httpwrapper error here.

Reply:
Good catch let me fix that